src/monitoring_tests/combinatorial_auction_surplus_test.py

```diff
@@ -0,0 +1,139 @@
+"""
+Comparing order surplus per token pair to a reference solver in the competition.
+"""
+# pylint: disable=logging-fstring-interpolation
+# pylint: disable=duplicate-code
+
+from typing import Any
+from fractions import Fraction
+from src.monitoring_tests.base_test import BaseTest
+from src.apis.web3api import Web3API
+from src.apis.orderbookapi import OrderbookAPI
+from src.models import Trade
+from src.constants import SURPLUS_ABSOLUTE_DEVIATION_ETH, SURPLUS_REL_DEVIATION
+
+
+class CombinatorialAuctionSurplusTest(BaseTest):
+    """
+    This test compares the surplus all orders from the winning settlement to
+    the different executions of these orders by other solvers in the competition.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.web3_api = Web3API()
+        self.orderbook_api = OrderbookAPI()
+
+    def compare_token_pairs_surplus(self, competition_data: dict[str, Any]) -> bool:
+        """
+        This function goes through each token pair that the winning solution executed
```

Comment:
Is it a directed token pair? I.e., sell_token = A and buy_token = B or {sell_token, buy_token} = {A,B}?